internal/endtoend/testdata/json_build/go/query.sql.go

```diff
@@ -0,0 +1,63 @@
+// Code generated by sqlc. DO NOT EDIT.
+// source: query.sql
+
+package querytest
+
+import (
+	"context"
+	"encoding/json"
+)
+
+const selectJSONBuildArray = `-- name: SelectJSONBuildArray :one
+SELECT 
+  json_build_array(1),
+  json_build_array(1, 2),
+  json_build_array(1, 2, 'foo'),
+  json_build_array(1, 2, 'foo', 4)
+`
+
+type SelectJSONBuildArrayRow struct {
+	JsonBuildArray   json.RawMessage
+	JsonBuildArray_2 json.RawMessage
+	JsonBuildArray_3 json.RawMessage
+	JsonBuildArray_4 json.RawMessage
+}
+
+func (q *Queries) SelectJSONBuildArray(ctx context.Context) (SelectJSONBuildArrayRow, error) {
+	row := q.db.QueryRowContext(ctx, selectJSONBuildArray)
+	var i SelectJSONBuildArrayRow
+	err := row.Scan(
+		&i.JsonBuildArray,
+		&i.JsonBuildArray_2,
+		&i.JsonBuildArray_3,
+		&i.JsonBuildArray_4,
+	)
+	return i, err
+}
+
+const selectJSONBuildObject = `-- name: SelectJSONBuildObject :one
+SELECT
+  json_build_object('foo'),
```

Comment:
I would add the following edge case to the tests this will likely fail.
```
json_build_object(),
```

Also I would remove the following tests because they actually fail in postgresql with the error
"ERROR:  argument list must have even number of elements"
```
  json_build_object('foo'),
  json_build_object('foo', 1, 2),
```
But catching those cases would be a little more difficult to catch without custom coding.